packages/components/src/public/QueryModel/GridPanel.tsx

```diff
@@ -253,28 +260,53 @@ interface GridTitleProps {
     allowSelections: boolean;
     allowViewCustomization: boolean;
     onRevertView?: () => void;
+    onSaveView?: () => void;
+    onSaveNewView?: () => void;
+    view?: ViewInfo;
+    isUpdated?: boolean
 }
 
 export const GridTitle: FC<GridTitleProps> = memo(props => {
-    const { title, model, onRevertView, actions, allowSelections, allowViewCustomization } = props;
+    const {
+        title,
+        view,
+        model,
+        onRevertView,
+        onSaveView,
+        onSaveNewView,
+        actions,
+        allowSelections,
+        allowViewCustomization,
+        isUpdated
+    } = props;
     const { queryInfo, viewName } = model;
 
     let displayTitle = title;
-    let view;
-    if (viewName) {
-        view = queryInfo.views.get(viewName.toLowerCase());
-        if (!view?.hidden) {
-            const label = view?.label ?? viewName;
-            displayTitle = displayTitle ? displayTitle + ' - ' + label : label;
+    let currentView = view;
+
+    if (!currentView) {
+        if (viewName) {
+            currentView = queryInfo.views.get(viewName.toLowerCase());
+        }
+        else {
+            currentView = queryInfo?.views.get(ViewInfo.DEFAULT_NAME.toLowerCase());
         }
-    } else {
-        view = queryInfo?.views.get(ViewInfo.DEFAULT_NAME.toLowerCase());
     }
-    const isEdited = view?.session && !view?.hidden;
-    const showSave = allowViewCustomization && isEdited && view?.savable;
-    const showRevert = allowViewCustomization && isEdited && view?.revertable;
+    if (!currentView?.hidden && viewName) {
+        const label = currentView?.label ?? viewName;
+        displayTitle = displayTitle ? displayTitle + ' - ' + label : label;
+    }
+
+    const isEdited = currentView?.session && !currentView?.hidden;
+    const showSave = allowViewCustomization && isEdited && currentView?.savable;
+    const showRevert = allowViewCustomization && isEdited && currentView?.revertable;
+    let canSaveCurrent = false;
+    if (!!viewName)
+        canSaveCurrent = true;
+    else if (isEdited)
+        canSaveCurrent = useServerContext().user.hasAdminPermission();
```

Comment:
Good reminder about guest. I tried the useMemo approach but I'm having a hard time getting jest tests to pass since now all consumer of GridTitle (including gridPanel) needs to have a serveContext. So I think will leave this as is.